internal/maintenance/job_cleaner.go

```diff
@@ -124,9 +124,11 @@ func (s *JobCleaner) Start(ctx context.Context) error { //nolint:dupl
 				continue
 			}
 
-			s.Logger.InfoContext(ctx, s.Name+logPrefixRanSuccessfully,
-				slog.Int("num_jobs_deleted", res.NumJobsDeleted),
-			)
+			if res.NumJobsDeleted > 0 {
+				s.Logger.DebugContext(ctx, s.Name+logPrefixRanSuccessfully,
+					slog.Int("num_jobs_deleted", res.NumJobsDeleted),
+				)
+			}
```

Comment:
There's some argument to be made that these changes are detrimental to debugging tests, but I generally find that if I'm debugging anything non-trivial I need to add more log statements anyway because the existing ones just aren't enough.